### ISSUE TITLE
Add SETTINGS parameter for index keys

### ISSUE THREAD

**Issue by porunov** [JanusGraph/janusgraph#1104]
I think it would be very useful to add `SETTINGS` parameter for index creation. With this option we could create more useful indexes like sorted index (https://www.elastic.co/blog/index-sorting-elasticsearch-6-0).

I imagine the creation process of sorted index could be like this (just a suggestion, I am not sure of it):
```
mgmt.buildIndex('borderIndex', Vertex.class).addKey(name, Settings.SORTED.asParameter())
.buildMixedIndex("search")
```

Related discussion:
https://groups.google.com/forum/#!searchin/janusgraph-users/elasticsearch$206%7Csort:date/janusgraph-users/dtBxqdR2qr0/ieXJdWaVAAAJ

**Comment by porunov** [comment]
Current workaround solution is to use Elasticsearch templates. 
1. Create a tamplate for JanusGraph indices like "janusgraph_*" (or more specific indices). In the template in settings I use sort and create mapping for sorted fields.
2. Use JanusGraph to create indices. JanusGraph will use next flow: create an index (sorting and base mapping will be applied by templates); make mappings.

**Comment by levilentz** [comment]
A feature such as this would be very beneficial for an enterprise deployment I am working on. Is this in the future roadmap for janusgraph?

**Comment by porunov** [comment]
@levilentz This feature isn't targeted 0.5.0 milestone of JanusGraph. 
As stated above, there is a workaround to configure necessary settings for indexes by using ElasticSearch directly (using ES templates for settings which should be applied during index creation or using settings update for settings which can be changed after the index creation).
If you have a desire to contribute this feature, it is more then welcome.
If you have no possibility to contribute this feature, I _may_ work on it but I have no plans to work on this feature in the near future due to the lack of interest and a small practical benefit of the feature.

**Comment by levilentz** [comment]
@porunov Yes, that workaround is what we are currently using as well, we just have some scaling concerns with the two-step nature of the index creation process. 

I think I may have some flexibility to help out with this after the new year as it would be a nice little addition.
